app/scripts/contentscript.js

```diff
@@ -82,6 +80,52 @@ function injectScript(content) {
   }
 }
 
+/**
+ * SERVICE WORKER LOGIC
+ */
+
+const WORKER_KEEP_ALIVE_INTERVAL = 1000;
+const WORKER_KEEP_ALIVE_MESSAGE = 'WORKER_KEEP_ALIVE_MESSAGE';
+const TIME_45_MIN_IN_MS = 45 * 60 * 1000;
+
+/**
+ * Don't run the keep worker alive logic for initial JSON RPC methods.
+ * We should not run keep-alive logic for non-dapp pages.
+ */
+const IGNORE_INIT_METHODS_FOR_KEEP_ALIVE = [
+  MESSAGE_TYPE.GET_PROVIDER_STATE,
+  MESSAGE_TYPE.SEND_METADATA,
+];
```

Comment:
@digiwand : can you plz detail why we are excluding initial JSON RPC methods here ?

Reply:
@jpuri 

just above these lines is the comment:
```
/**
 * Don't run the keep worker alive logic for initial JSON RPC methods.
 * We should not run keep-alive logic for non-dapp pages.
 */
 ```
 
 is there something you could recommend me to add or update?

Reply:
I do not understand `initial JSON RPC methods` part.

Reply:
@jpuri 

I see. good callout. what about this [073f05e](https://github.com/MetaMask/metamask-extension/pull/16075/commits/073f05ec972fd9f09aee24f1302d2f566c2bc3fc)?